Automattic/jetpack
# My Jetpack: Add async site connection to site connection Notice

<img width="1173" alt="image" src="https://github.com/Automattic/jetpack/assets/5714212/2dfadeb7-3aa7-4bdb-aa20-cc9ddef406aa">
<img width="1173" alt="image" src="https://github.com/Automattic/jetpack/assets/5714212/0d544ea2-78e8-47bc-8d4d-11ff92cf7cb9">
<img width="1173" alt="image" src="https://github.com/Automattic/jetpack/assets/5714212/bafdc06e-5ad2-4625-98b8-df443ebd0dea">


See linked issue

## Proposed changes:

* This PR updates the site connection button to perform an async site connection, instead of directing to the user connection screen. After a user clicks on the button, the button will get into a loading state and the site will try to connect to WordPress.com in the background. After the connection, it will add a confirmation notice, stating that the site has been successfully connected.


### Other information:


## Jetpack product discussion


p1HpG7-qKY-p2

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* Spin up a JN site with Jetpack Beta pointing to this branch
* Visit `/wp-admin/admin.php?page=my-jetpack`
* Dismiss the Welcome banner and refresh the page
* You'll now see a banner saying "Some products need a connection to WordPress.com to be able to work.". Click on "Connect your site to fix this"
* The label will change to "Connecting..."
* After that loading, you'll see a success notice with the following message: "Your site has been successfully connected."

NOTE: error handling and the state of the cards will be handled in a different PR
